src/NuGet.Clients/VisualStudio/Extensibility/IVsPackageProjectMetadata.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace NuGet.VisualStudio
+{
+    /// <summary>
+    /// Contains information about project where we're executing nuget actions.
+    /// </summary>
+    [ComImport]
+    [Guid("7ce0caae-1b99-4cd1-80bf-348972975bcf")]
+    public interface IVsPackageProjectMetadata
+    {
+        /// <summary>
+        /// Batch id for batch event of the project
```

Comment:
This should mention that each batch is unique, and that start/end will contain the same. Just to make this painfully clear to callers